packages/perseus/src/widgets/interactive-graphs/graphs/use-draggable.test.tsx

```diff
@@ -127,21 +130,55 @@ describe("useDraggable", () => {
                 <TestDraggable
                     point={[0, 0]}
                     onMove={onMoveSpy}
-                    constrain={(p) => [Math.round(p[0]), Math.round(p[1])]}
+                    constrainKeyboardMovement={(p) => snap([1, 1], p)}
                 />
             </Mafs>,
         );
         const dragHandle = screen.getByRole("button");
 
-        // Act: click and hold the drag handle...
-        mouseDownAt(dragHandle, 0, 0);
-        // ...and then drag 12px right and 13px down
-        moveMouseTo(dragHandle, 12, 13);
+        // Act
+        await userEvent.tab();
+        await userEvent.type(dragHandle, "{arrowright}");
 
-        // Assert: the draggable element was moved to (1, -1) due to the
-        // constrain function. If you see (1.2, -1.3) instead, that means the
-        // constraint is not being applied.
-        expect(onMoveSpy).toHaveBeenCalledWith([1, -1]);
+        // Assert: the draggable element was moved one step to the right
+        expect(onMoveSpy.mock.calls).toEqual([[[1, 0]]]);
+    });
+
+    it("constrains the destination point using a constrainKeyboardMovement object", async () => {
+        // Arrange: a 200x200px graph with a 20-unit range in each dimension.
```

Comment:
I really appreciate these comment explanations about what the graph is setting up!